tests/helpers.py

```diff
@@ -133,13 +133,26 @@ def create_pod(
         AssertionError: In case of any failure
     """
     if interface_type == constants.CEPHBLOCKPOOL:
-        pod_dict = pod_dict_path if pod_dict_path else constants.CSI_RBD_POD_YAML
         interface = constants.RBD_INTERFACE
+        if pod_dict_path:
+            pod_dict = pod_dict_path
+        else:
+            pod_dict = constants.CSI_RBD_POD_YAML
+            pull_images('nginx')
     else:
-        pod_dict = pod_dict_path if pod_dict_path else constants.CSI_CEPHFS_POD_YAML
         interface = constants.CEPHFS_INTERFACE
+        if pod_dict_path:
+            pod_dict = pod_dict_path
+        else:
+            pod_dict = constants.CSI_CEPHFS_POD_YAML
+            pull_images('nginx')
+
     if dc_deployment:
-        pod_dict = pod_dict_path if pod_dict_path else constants.FEDORA_DC_YAML
+        if pod_dict_path:
+            pod_dict = pod_dict_path
+        else:
+            pod_dict = constants.FEDORA_DC_YAML
+            pull_images('fedora')
```

Comment:
We have a little bit of duplication here that we can reduce. It seems like in every conditional if we specify `pod_dict_path` then that takes priority. If this is the case we can specify logic first and then if it isn't set we can check our other conditionals:

```
if pod_dict_path:
    pod_dict = pod_dict_path
else:
    if interface_type == constants.CEPHBLOCKPOOL:
        interface = constants.RBD_INTERFACE
        pod_dict = constants.CSI_RBD_POD_YAML
        pull_images('nginx')
    else:
        interface_type = constants.CEPHFS_INTERFACE
        pod_dict = constants.CSI_CEPHFS_POD_YAML
        pull_images('nginx')
    if dc_deployment:
        pod_dict = constants.FEDORA_DC_YAML
        pull_images('fedora')

```

One thing to note here is that I kept the same pattern of if/else/if. I'm not sure if this is intentional or not but if we  have an `interface_type` of `constants.CEPHBLOCKPOOL` and `dc_deployment` is true, we will overwrite the values for `CEPHBLOCKPOOL` and instead use the ones for `dc_deployment` since it is not an `elif`. Again, I'm not sure if this is intentional so I left it as is.

Reply:
Thanks Coady. I thought about this part i.e., to have a check for `pod_dict_path` before sending this PR. 

I thought of something like this,

```python
    if not pod_dict_path:
        if interface_type == constants.CEPHBLOCKPOOL:
            interface = constants.RBD_INTERFACE
            pod_dict = constants.CSI_RBD_POD_YAML
            pull_images('nginx')
        elif interface_type == constants.CEPHFS_INTERFACE:
            interface = constants.CEPHFS_INTERFACE
            pod_dict = constants.CSI_CEPHFS_POD_YAML
            pull_images('nginx')
    else:
        pod_dict = pod_dict_path
        if interface_type == constants.CEPHBLOCKPOOL:
            interface = constants.RBD_INTERFACE
        elif interface_type == constants.CEPHFS_INTERFACE:
            interface = constants.CEPHFS_INTERFACE
```

The reason I sent it in the current form is that the interface type needs to be set irrespective of whether `pod_dict_path` is defined or not. `pod_dict_path` doesn't define the interface type. So,  checking the `interface type` made more sense to me. pls, let me know what you think.

Reply:
Yes that's a good point and something I missed. Your suggestion works for me!

Reply:
Thanks. Resolving the comment.